File: composeApp/src/desktopMain/kotlin/org/queststudios/yamlvalidation/validation/ValidationRule.kt
package org.queststudios.yamlvalidation.validation

import org.queststudios.yamlvalidation.core.ValidationContext

interface ValidationLogger {
    fun log(level: String, message: String)
}

interface ValidationRule {
    fun validate(endpoint: String, method: String, context: ValidationContext, logger: ValidationLogger)
    val name: String
}

File: src/validation/ValidationRule.js
export class ValidationLogger {
  log(level, message) {
    // To be implemented by concrete logger
  }
}

export class ValidationRule {
  get name() {
    return this.constructor.name;
  }

  validate(endpoint, method, context, logger) {
    // To be implemented by concrete rules
    throw new Error('validate() must be implemented by subclass');
  }
}

